Lib/concurrent/futures/_base.py

```diff
@@ -170,6 +170,17 @@ def _create_and_install_waiters(fs, return_when):
 
     return waiter
 
+
+def _yield_future(fs, waiter, ref_collect=()):
+    while fs:
+        with fs[0]._condition:
```

Comment:
If I'm understanding correctly then, yes, I would certainly consider it an error to call `set_result` twice.